src/cli.rs

```diff
@@ -0,0 +1,213 @@
+use crate::merkle_tree::MerkleTree;
+use std::error::Error;
+use std::{io, vec};
+
+/// The `CLI` struct is used to manage the command line interface of the Merkle Tree.
+pub struct CLI {
```

Comment:
Have you considered using [clap](https://docs.rs/clap/latest/clap/) for the CLI?

Reply:
This is so useful thanks! I'll look into it :)

Reply:
Done!